### PR TITLE
Fix hljs highlighting for safari browser

### PR DESCRIPTION
**Pull request by BarkingBad**

### CODE REVIEW

**Comment by BarkingBad** [comment]
Before:
obraz
After:
obraz

A tried to change regexes to be `look behind` free as safari browser doesn't support that yet. I used a trick proposed [here](https://stackoverflow.com/questions/51568821/works-in-chrome-but-breaks-in-safari-invalid-regular-expression-invalid-group/51568859). Could you @TheElectronWill look into them whether they are correct?

File: scaladoc/resources/dotty_res/scripts/hljs-scala3.js [hunk 1]
```diff
@@ -36,7 +36,7 @@ function highlightDotty(hljs) {
   function titleFor(name) {
     return {
       className: 'title',
-      begin: `(?<=${name} )${id.source}`
+      begin: `((?:${name} )(${id.source})`
```
Comment (TheElectronWill) [inline scaladoc/resources/dotty_res/scripts/hljs-scala3.js:39]: missing parenthesis?

**Comment by TheElectronWill** [comment]
I'm afraid this simple fix won't work.

- changing `id` this way highlights the method names incorrectly: in `def func_=` the `=` will not be included in the function's name  
- changing `titleFor` breaks the highlighting of "titles", i.e. the names of variables and methods aren't highlighted anymore.
example:
image
becomes
image

- changing `ENUM_CASE` changes how cases are detected, they end up as `type` instead of `title`. And in `case A, B`, `A` and `B` won't be labelled the same way. 
- changing the markdown links this way includes the label into the url, which is a bit weird semantically. If you underline the links you see the difference : 
  image becomes image

**Review by TheElectronWill — CHANGES_REQUESTED** [review]
There should be a better way of avoiding the look-behinds. I'll try some workarounds :slightly_smiling_face: 
[Related discussion in hljs repo](https://github.com/highlightjs/highlight.js/issues/2520)

**Comment by BarkingBad** [comment]
> I'll try some workarounds

Would you like to overtake that issue?

**Comment by TheElectronWill** [comment]
> > I'll try some workarounds
> 
> Would you like to overtake that issue?

I'll be glad to do it! I'll assign myself

**Comment by BarkingBad** [comment]
No problem, thank you for your support :D 

I think we can close that PR since it's on my fork and it would be hard to cooperate, and when you are ready, you'll open a fresh new one. I'll assign you to the related issue #11771

**Pull request closed without merging**